index.js

```diff
@@ -74,6 +79,10 @@ module.exports = (dest, opts) => {
 			newPath = gutil.replaceExtension(newPath, opts.extension);
 		}
 
+		if (opts.transform) {
+			newPath = opts.transform(newPath) || '';
```

Comment:
But what if `opts.transorm(path)` returns a number of an object or a boolean? Maybe we should really type check this and throw an error if it doesn't return a string.